.circleci/config.yml

```diff
@@ -1,38 +1,77 @@
 version: 2.1
 
 orbs:
-  orb-tools: circleci/orb-tools@9.0.0
+  orb-tools: circleci/orb-tools@9.0.1
   golang-ci: financial-times/golang-ci@<<pipeline.parameters.dev-orb-version>>
 
+tags_and_branches: &tags_and_branches
+  tags:
+    only: /^(major|minor|patch)-release-v\d+\.\d+\.\d+$/
+
+only_tags: &only_tags
+  branches:
+    ignore: /.*/
+  tags:
+    only: /^(major|minor|patch)-release-v\d+\.\d+\.\d+$/
+
+only_branches: &only_branches
+  branches:
+    only: /.*/
+
 parameters:
   dev-orb-version:
-    default: 'dev:alpha'
+    default: dev:alpha
     type: string
+  run-integration-tests:
+    default: false
+    type: boolean
 
 workflows:
-  lint_pack-validate_publish-dev:
+  lint_publish-dev_prod-release:
+    unless: << pipeline.parameters.run-integration-tests >>
     jobs:
-      - orb-tools/lint
+      - orb-tools/lint:
+          filters: *tags_and_branches
 
       - orb-tools/pack:
           requires:
             - orb-tools/lint
+          filters: *tags_and_branches
 
       - orb-tools/publish-dev:
           orb-name: financial-times/golang-ci
-          context: publish-orbs
-          publish-sha-version: false
+          context: cm-team-orb-publishing
           requires:
             - orb-tools/pack
+          filters: *tags_and_branches
 
-      # the following jobs act as integrations tests for this orb, we run them and expect not to error
-      # they should use the just published dev:alpha version
-      - golang-ci/build-and-test:
-          name: default-integration-tests
+      - orb-tools/trigger-integration-tests-workflow:
+          context: cm-team-orb-publishing
           requires:
             - orb-tools/publish-dev
+          filters: *only_branches
 
-      - golang-ci/docker-build:
-          name: docker-job-integration-tests
+      # This step is executed only on tag release. Please note that currently CircleCI is not able to execute
```

Comment:
Why is it not possible to run integration tests on tag creation?

Reply:
In order to run integration tests on the latest orb (deployed in dev:alpha), the integration tests should be run in new pipeline started by the previous pipeline programatically. Unfortunately the second pipeline was not triggered when I made the correct filters to be runned from tag. I couldn't find solution on that problem(pipeline triggered by another pipeline that is triggered by tag creation) and that's why I mention in the README that if someone finds a way, contribution is welcome.